apps/mobile-wallet/src/persistent-storage/wallet.ts

```diff
@@ -273,6 +279,15 @@ export const getStoredWalletMetadata = async (error?: string): Promise<WalletMet
   return walletMetadata
 }
 
+export const isStoredWalletMetadataMigrated = (
```

Comment:
It's confusing to me that we mix `walletMetada` and `addressMetadata` like that. Is that correct?

This feels wrong:
- `addressMetadata as WalletMetadata`
- `addressMetadata is WalletMetadata`